app/components/provider_interface/change_course_details_component.rb

```diff
@@ -37,30 +39,21 @@ def rows
     end
 
     def preferred_location_text
-      "#{application_choice.site.name_and_code}\n" \
-        "#{formatted_address}"
+      course_option.site.name_and_address
     end
 
     def accredited_body
-      accredited_body = @application_choice.course.accredited_provider
+      accredited_body = course.accredited_provider
       accredited_body.present? ? accredited_body.name_and_code : provider_name_and_code
     end
 
     def funding_type
-      key = @application_choice.course.funding_type.to_sym
+      key = course.funding_type.to_sym
       FUNDING_TYPES[key].to_s.humanize
     end
 
   private
 
-    def formatted_address
```

Comment:
since we are going to do the changes later can we maintain the previous logic as the read only version is currently live and this will change the behaviour

Reply:
ok!